manifests/init.pp

```diff
@@ -64,8 +67,9 @@
   validate_hash($settings)
   validate_hash($ppas)
 
-  if $proxy['host'] {
+  if $proxy['ensure'] or $proxy['host'] {
```

Comment:
I think we'd need to use `$_proxy['ensure']` instead of `$proxy['ensure']` to make sure it doesn't break strict variables, but otherwise yes, I think that should work.